### PR TITLE
Add a call of assert_vars() for both min_dates and max_dates to check…

### PR DESCRIPTION
**Pull request by hamzarahal**
… that must be a list of variable names

### CODE REVIEW

File: R/derive_date_vars.R [hunk 1]
```diff
@@ -134,7 +134,7 @@
 #' # minimum dates
 #' impute_dtc(
 #'   "2020-12",
-#'   min_dates = list(
+#'   min_dates = vars(
 #'     ymd_hms("2020-12-06T12:12:12"),
 #'     ymd_hms("2020-11-11T11:11:11")
 #'   ),
```
Comment (thomas-neitmann) [inline R/derive_date_vars.R:140]: Only make the change when `derive_vars_dt()` or `derive_vars_dtm()` is called but **not** for `impute_dtc()`.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: R/derive_date_vars.R [hunk 2]
```diff
@@ -686,15 +691,15 @@ derive_vars_dtm <- function(dataset,
...
-  )
+  dataset <- dataset %>%
+    mutate(
+      !!sym(dt) := convert_dtc_to_dtm(
+        dtc = !!dtc,
+        date_imputation = date_imputation,
+        min_dates = lapply(min_dates, eval_tidy, data = rlang::as_data_mask(.)),
+        max_dates = lapply(max_dates, eval_tidy, data = rlang::as_data_mask(.))
+      )
+    )
```
[the remark this answers is not in the document]
Comment (hamzarahal) [reply R/derive_date_vars.R:702]: Updated for derive_vars_dtm() code

**Comment by hamzarahal** [comment]
@thomas-neitmann may need to check again for the related vignettes and see if the updates are well implemented? because the PR fail is still coming from vignettes

**Comment by hamzarahal** [comment]
@thomas-neitmann  I have updated for both test-impute_dtc.R and imputation.Rmd to change vars by list when we call the min/max dates parameters for impute_dtc() call, also checked for the ones using impute_dtc that list() is present whenever we call min/max dates,

**Comment by thomas-neitmann** [comment]
Thanks @hamzarahal! There's still a case where `vars()` is being used inside `impute_dtc()`, namely in the example of that function. 
image

**Comment by hamzarahal** [comment]
@thomas-neitmann That means that the issue is still coming from man/imputec_dtc, since when checking derive_date_vars() the update is implemented yet, if it's the case need to commit/push only that file?

Capture d’écran 2021-11-09 161258

**Comment by thomas-neitmann** [comment]
You can to run `devtools::document()` and then commit the changes file in the `man` folder.

**Review by thomas-neitmann — APPROVED** [review]
Thanks!

**Pull request merged**